### PR TITLE
Add vector catalog UI

### PR DESCRIPTION
**Pull request by kdp-cloud**
This PR is for the implementation of adding a Level 0 and a backbone to the table in the catalogue page of the UI.
Further, there is also an elm-file (Accordion.elm) for the implementation of an accordion pane in the vector catalogue.

### CODE REVIEW

File: client/index.html [hunk 1]
```diff
@@ -11,16 +12,19 @@
 			margin: 0;
 		}
 	</style>
+	<script src="https://unpkg.com/elm-accordion@1.0.4/dist/elm-accordion.min.js"></script>
```
Comment (MaybeJustJames) [inline client/index.html:15]: Remove

File: client/index.js [hunk 2]
```diff
@@ -1,3 +1,6 @@
 import { Elm } from "./src/Main.elm";
 
-Elm.Main.init({ node: document.getElementById("elm") });
\ No newline at end of file
+// require("elm-accordion/dist/elm-accordion.min.js");
+// require("elm-accordion/dist/elm-accordion.min.css");
+
```
Comment (MaybeJustJames) [inline client/index.js:5]:
```suggestion
```

File: client/package.json [hunk 3]
```diff
@@ -13,5 +13,8 @@
     "elm": "^0.19.1-5",
     "elm-live": "^4.0.2",
     "parcel": "^2.0.0"
+  },
+  "dependencies": {
+    "elm-accordion": "^1.0.4"
```
Comment (MaybeJustJames) [inline client/package.json:18]:
```suggestion
```

**Review by MaybeJustJames — CHANGES_REQUESTED** [review]
Some initial issues

**Comment by kdp-cloud** [comment]
This PR is for the implementation of adding a Level 0 and a backbone to the table in the catalogue page of the UI.
Further, there is also an elm-file (Accordion.elm) for the implementation of an accordion pane in the vector catalogue.

File: client/src/Main.elm [hunk 4]
```diff
@@ -236,7 +341,240 @@ type Msg
...
+                            [ Input.text []
+                                { onChange = FilterBackboneTable
+                                , text = Maybe.withDefault "" model.backboneFilterString
+                                , label = Input.labelLeft [] <| Element.text "Filter:"
+                                , placeholder = Nothing
+                                }
+                            , backboneTable model
+                            , Element.row
+                                [ centerX, spacing 50 ]
+                                [ customAddButton "+" (SwitchPage AddBackbonePage) ]
```
Comment (MaybeJustJames) [inline client/src/Main.elm:420]: `customAddButton` is undefined

File: client/src/Main.elm [hunk 5]
```diff
@@ -236,7 +341,240 @@ type Msg
...
+                        (Accordion.body [spacing 25]
+                            [ Input.text []
+                                { onChange = FilterLevel0Table
+                                , text = Maybe.withDefault "" model.level0FilterString
+                                , label = Input.labelLeft [] <| Element.text "Filter:"
+                                , placeholder = Nothing
+                                }
+                            , overhangRadioRow model
+                            , insertTable model
+                            , Element.row [ centerX, spacing 50 ] [ customAddButton "+" (SwitchPage AddLevel0Page) ]
```
Comment (MaybeJustJames) [inline client/src/Main.elm:444]: `customAddButton` is undefined

**Review by MaybeJustJames — CHANGES_REQUESTED** [review]
3 build errors:
* `customAddButton` is undefined (420 and 444)
* Line 797 uses an ambiguous `text`

**Comment by MaybeJustJames** [comment]
Also make sure your `package-lock.json` is up to date (`npm i` will do it)

**Review by MaybeJustJames — APPROVED** [review]
Make sure you squash some of these commits before you merge. Great work :)

**Pull request merged**